bridge_tool_cli.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+
+"""Example of using WyzeSense USB bridge.
+
+**Usage:** ::
+  bridge_tool_cli.py [options]
+
+**Options:**
+
+    -d, --debug     output debug log messages to stderr
+    -v, --verbose   print and log more information
+    --device PATH   USB device path [default: /dev/hidraw0]
+
+**Examples:** ::
+
+  bridge_tool_cli.py --device /dev/hidraw0
+
+"""
+from __future__ import print_function
+
+from builtins import input
+
+import re
+import sys
+import logging
+import binascii
+import wyzesense
+
+
+def on_event(ws, e):
+    s = "[%s][%s]" % (e.Timestamp.strftime("%Y-%m-%d %H:%M:%S"), e.MAC)
+    if e.Type == 'state':
+        (s_type, s_state, s_battery, s_signal) = e.Data
+        s += f"StateEvent: sensor_type={s_type}, state={s_state}, " \
+             f"battery={s_battery}, signal={s_signal}"
+    else:
+        s += "RawEvent: type=%s, data=%r" % (e.Type, e.Data)
+    print(s)
+
+
+def main(args):
+    if args['--debug']:
+        loglevel = logging.DEBUG - (1 if args['--verbose'] else 0)
+        logging.getLogger("wyzesense").setLevel(loglevel)
+        logging.getLogger().setLevel(loglevel)
+
+    device = args['--device']
+    print("Openning wyzesense gateway [%r]" % device)
+    try:
+        ws = wyzesense.Open(device, on_event)
+        if not ws:
+            print("Open wyzesense gateway failed")
+            return 1
+        print("Gateway info:")
+        print("\tMAC:%s" % ws.MAC)
+        print("\tVER:%s" % ws.Version)
+        print("\tENR:%s" % binascii.hexlify(ws.ENR))
+    except IOError:
+        print("No device found on path %r" % device)
+        return 2
+
+    def List(unused_args):
+        result = ws.List()
+        print("%d sensor paired:" % len(result))
+        logging.debug("%d sensor paired:", len(result))
+        for mac in result:
+            print("\tSensor: %s" % mac)
+            logging.debug("\tSensor: %s", mac)
+
+    def Pair(unused_args):
+        result = ws.Scan()
+        if result:
+            print("Sensor found: mac=%s, type=%d, version=%d" % result)
+            logging.debug("Sensor found: mac=%s, type=%d, version=%d", *result)
+        else:
+            print("No sensor found!")
+            logging.debug("No sensor found!")
+
+    def Unpair(mac_list):
+        for mac in mac_list:
+            if len(mac) != 8:
+                print("Invalid mac address, must be 8 characters: %s", mac)
+                logging.debug("Invalid mac address, must be 8 characters: %s",
+                              mac)
+                continue
+
+            print("Un-pairing sensor %s:" % mac)
+            logging.debug("Un-pairing sensor %s:", mac)
+            ws.Delete(mac)
+            print("Sensor %s removed" % mac)
+            logging.debug("Sensor %s removed", mac)
+
+    def Fix(mac_list):
```

Comment:
The Fix method is doing to same thing as Unpair, but with a specific value for mac that shows up when a sensor fails and stops reporting it's true mac. Unfortunately seemed to be a common issue I was seeing in forums. Since the Unpair function has code to protect from sending an invalid mac, I created the second function to just hard code and not validate the mac. This also keeps the rest of the script as close to the source that I borrowed it from.

Reply:
Differentiated the Fix and Unpair functions more, including a better way to handle multiple invalid mac addresses that have been found. Also converted script to use f-string syntax and some other minor tweaking. Resolving this.